utils/converter_utils.py

```diff
@@ -166,25 +161,28 @@ def attrib2dict(ob):
     return attrib_dict
 
 
-def get_sdrf_path(idf_file_path, logger):
+def get_sdrf_path(idf_file_path, logger, data_dir):
     """Read IDF and get the SDRF file name, look for the SDRF in the data directory (i.e. "unpacked")
     or in the same directory as the IDF.
 
     :param idf_file_path: full or relative path to IDF file
     :param logger: log handler
+    :param data_dir: path to folder with SDRF
     :return: path to SDRF file
     """
 
     current_dir = os.path.dirname(idf_file_path)
     sdrf_file_path = ""
+    if not data_dir:
+        data_dir = DEFAULT_DATA_DIRECTORY
     # Figure out the name and location of sdrf files
     with codecs.open(idf_file_path, 'rU', encoding='utf-8') as f:
         # U flag makes it portable across in unix and windows (\n and \r\n are treated the same)
         for line in f:
             if re.search(SDRF_FILE_NAME_REGEX, line):
                 sdrf_file_name = line.split('\t')[1].strip()
-                if os.path.exists(current_dir + DATA_DIRECTORY):
-                    sdrf_file_path = os.path.join(current_dir, DATA_DIRECTORY, sdrf_file_name)
+                if os.path.exists(os.path.join(current_dir, data_dir)):
```

Comment:
suggest defining a variable to hold os.path.join(current_dir, data_dir), then using that in the sdrf_file_path line. Less likely to develop a bug due to mismatches in future ;-).

Reply:
Thanks! Done.